core/base/src/main/java/de/mm20/launcher2/helper/IterableUtils.kt

```diff
@@ -0,0 +1,10 @@
+package de.mm20.launcher2.helper
+
+object IterableUtils {
+    inline fun <T, R : Comparable<R>> List<T>.medianBy(crossinline selector: (T) -> R?): T? {
+        if (this.isEmpty())
+            return null
+
+        return this.sortedBy { selector(it) } [ this.size / 2 ]
+    }
+}
```

Comment:
1. This should be moved to a file called `Iterable.kt` in the `:core:ktx` module.
2. It doesn't need to be wrapped in an object.
3. I assume it could be generalized to `Iterable<T>.medianBy` instead of `List<T>.medianBy`